src/renderers/opengl/mod.rs

```diff
@@ -89,23 +121,86 @@ impl GlCache {
     }
 }
 
-type ErasedNodeRenderer = Box<dyn Fn(&OpenglRenderer, &dyn Node)>;
+pub type OpenglRenderer = ExtOpenglRenderer<(), DefaultCustomRenderer>;
 
-pub struct OpenglRenderer {
+pub trait CustomRenderer {
+    type NodeData;
+    type Renderer;
+
+    fn render(
+        &self,
+        renderer: &Self::Renderer,
+        node: &ExtInoxNode<Self::NodeData>,
+        node_data: &Self::NodeData,
+    );
+}
+
+pub struct DefaultCustomRenderer;
+
+impl CustomRenderer for DefaultCustomRenderer {
+    type NodeData = ();
+    type Renderer = OpenglRenderer;
+
+    fn render(
+        &self,
+        _renderer: &Self::Renderer,
+        _node: &ExtInoxNode<Self::NodeData>,
+        _node_data: &Self::NodeData,
+    ) {
+    }
+}
+
+pub fn opengl_renderer(
+    gl: glow::Context,
+    nodes: InoxNodeTree,
+    textures: Vec<ModelTexture>,
+) -> OpenglRenderer {
+    ExtOpenglRenderer::new(gl, nodes, textures, DefaultCustomRenderer)
+}
+
+pub fn opengl_renderer_ext<T, R>(
```

Comment:
Is `opengl_renderer_ext` intended for the Khronos spec with Extensions and  `opengl_renderer` intended for the official Khronos spec API?

Reply:
But I didn't see any Extension APIs being used?

Reply:
Khronos spec? I don't know what this is.

No, `opengl_renderer_ext` is made for people who make custom nodes and want to render them somehow. For example an SVG node or a Lua script node.

Reply:
Ohhh I seee